### PR TITLE
Misleading visual representation of links #25

### LINKED ISSUE #25: Misleading visual representation of links
3unnamed
Colours used in links aren't very good.
Blue is the recognised colour for links, but external links are just black and underlined in the content.
Internal links (the blue ones) loads content via AJAX, but it's not clear that it's a link either because hovering over it gives you a vertical bar instead of the usual hand.
Some sort of indicator for AJAX-loading-in-progress would help too; clicking on those give no immediate indication anything even happened.

### PR DESCRIPTION
**Pull request by vhazali**
Fixes #25 
[Preview](http://vhazali.github.io/website)

### CODE REVIEW

**Comment by damithc** [comment]
The location of the 'loading' gif can be improved. Right now it appears on top of existing content. Instead, it should be at the vertical top of the area where new content is going to appear?
I'm not sure if we should change the link color though. Blue is the common color for links and there is no common color for embedded links. We can think of another way to distinguish between normal links and embedded links. May be use dashed underline? Anyway, that can be done in a separate issue too.

**Comment by jkt001** [comment]
I have a few suggestions:

What I tried to do before was to make it look like this:

screen shot 2015-12-31 at 19 44 29

and when a user hovers over it, it turns dark blue (and the cursor becomes a hand):

screen shot 2015-12-31 at 19 44 51

I don't know if this approach is good, but it's a start. As for dashed underline, in my opinion, it is better for words that have tooltips #31 that are dashed-underline. 

One of my concern for my approach is that it might be confusing with the yellow highlighting as mentioned in #23. Since this is a design type issue, there might be more than one opinion on this. Any other opinion / thoughts are much appreciated.

**Comment by damithc** [comment]
I agree about the dash underline. Let's keep it as before (blue text, underlined, just like a normal link). At a later time, we can add a tiny icon to distinguish embedded links.

File: styles/common.css [hunk 1]
```diff
@@ -367,8 +377,14 @@ table.bordered tr > th {
  */
 
 .embedded-link {
-    color: blue;
+    color: green;
     text-decoration: underline;
+    cursor: pointer;
+    cursor: hand;
```
Comment (wkurniawan07) [inline styles/common.css]: is this double definition of `cursor` intended?

File: scripts/common.js [hunk 2]
```diff
@@ -7,6 +7,7 @@ function makeAccordion(elementSelector) {
 }
 
 function getContentUsingAjax(section, elementSelector) {
+		$("#embedded-link-loading-img").show();
```
Comment (wkurniawan07) [inline scripts/common.js]: Any reason why this is put here? Why not in `pullContent` right after `success`, it flows more logically as well.

**Comment by wkurniawan07** [comment]
Take a look at the indentation. Use spaces, not tabs.

**Comment by wkurniawan07** [comment]
Regarding the solution with blue highlight, I don't really suggest it because from HCI point of view highlights are supposedly representing something important, not merely something different.

File: styles/common.css (hunk 1 above)
Comment (vhazali) [inline styles/common.css]: Yup! For some reasons that I cannot explain, it doesn't work on my chrome if I just put in ```cursor: hand;``` even though some slight googling says that it should

File: scripts/common.js (hunk 2 above)
Comment (vhazali) [inline scripts/common.js]: I wasn't sure if we only wanted to show loading when ```getContentUsingAjax``` was called. I'll edit and move it down to ```pullContent```

**Comment by vhazali** [comment]
Hi all, sorry for the slow reply.

I'll work on the positioning of the image.
As for the colour of the ajax links, should I leave it as green? Or will it be better if I revert it to the original blue?

**Comment by damithc** [comment]
Revert to the original blue.

File: styles/common.css [hunk 3]
```diff
@@ -73,6 +73,16 @@ a:visited {
...
 
+#embedded-link-loading-img {
+  display: none;
+  position: fixed;
+  top: 50%;
+  left: 50%;
+  z-index: 11;
+  background-color: white;
+  font: 1.5em;
+}
```
Comment (wkurniawan07) [inline styles/common.css]: 4 spaces of indentation, not 2.

File: styles/common.css [hunk 4]
```diff
@@ -369,6 +379,12 @@ table.bordered tr > th {
 .embedded-link {
     color: blue;
     text-decoration: underline;
+    cursor: pointer;
+    cursor: hand;
+}
+
+.embedded-link:active {
+    color:purple;
```
Comment (wkurniawan07) [inline styles/common.css]: `color: purple;`

File: contents/schedule.html [hunk 5]
```diff
@@ -7,6 +7,7 @@
         <link type="text/css" rel="stylesheet" href="../styles/common.css" />
     </head>
     <body>
+        <img id="embedded-link-loading-img" src="/images/ajax-preload.gif" alt="Loading..."/>
```
Comment (wkurniawan07) [inline contents/schedule.html]: Two issues:
1. Don't close the `img` tag with `/>`.
2. The reference is broken. It should be `../images/ajax-preload.gif`. `common.js` is making the same mistake (but don't fix it here).

File: styles/common.css (hunk 1 above)
Comment (wkurniawan07) [inline styles/common.css]: What if you just put in `cursor: pointer;`? If it still fails, then I guess we have no choice but to include both.

File: styles/common.css [hunk 6]
```diff
@@ -369,6 +379,12 @@ table.bordered tr > th {
 .embedded-link {
     color: blue;
     text-decoration: underline;
+    cursor: pointer;
+    cursor: hand;
```
Comment (wkurniawan07) [inline styles/common.css]: You mentioned that putting in only `cursor: hand;` fails. What if you put in only `cursor: pointer;`? If it still fails, then I guess we have no choice but to include both.

**Comment by wkurniawan07** [comment]
Some minor changes and we're good to go if @damithc agrees with the solution :+1:

**Comment by damithc** [comment]
When I check the staging site, the 'loading gif' never shows up. All I see is the default 'image missing' placeholder. Is the image missing? Or the loading of the 'loading gif' takes too long (how ironic :-p)

**Comment by damithc** [comment]
Is the gif problem due to #146 ?

**Comment by vhazali** [comment]
@damithc yes I think it's due to the '''src''' tag as pointed out by @wkurniawan07 

I'll get to fixing the problems in awhile! 

I still can't find a good way for the loading gif to appear right under the link that's clicked. Might need a bit more time for that

**Comment by damithc** [comment]
Fine tuning the position can be done as a separate issue after this. Just create a new issue for it (and for any other potential improvements) after this one is merged.

File: styles/common.css (hunk 6 above)
Comment (vhazali) [inline styles/common.css]: Using just ```cursor: pointer;``` works!

**Comment by wkurniawan07** [comment]
Good to go :+1:

**Pull request merged**